dvc/stage_cache.py

```diff
@@ -0,0 +1,72 @@
+import os
+import json
+import hashlib
+
+from dvc.utils.fs import makedirs
+
+
+def _sha256(string):
```

Comment:
Decided to go with sha256 over md5 just to avoid people getting scared by non-cryptographic md5 :slightly_smiling_face: